pyro/infer/mcmc/util.py

```diff
@@ -469,6 +469,20 @@ def summary(samples, prob=0.9, num_chains=1):
     print('\n')
 
 
+def _predictive_sequential(model, posterior_samples, model_args, model_kwargs,
+                           num_samples, return_trace=False, sample_sites=None):
+    traces = []
+    samples = [{k: v[i] for k, v in posterior_samples.items()} for i in range(num_samples)]
+    for i in range(num_samples):
+        traces.append(poutine.trace(poutine.condition(model, samples[i])).get_trace(*model_args, **model_kwargs))
```

Comment:
I think that in case of `return_trace=False`, we don't want to hold all the traces in a list. For example, consider a model
```
def model(X, y):
    coef = pyro.sample('coef', dist.Normal(0, 1))
    pyro.sample('y', dist.Normal(coef * X, 1), obs=y)
```
We have a new tensor `coef * X` in each trace, which will cause memory problem if X is high dimensional and num_samples is large.

Reply:
Good point. Let me change this so that we start stacking samples right away if `return_trace=False.`